src/ml-search.service.js

```diff
@@ -1225,6 +1233,77 @@
     },
 
     /**
+     * Gets aggregates for facets (from a search response object) based on facet type
+     * @method MLSearchContext#getAggregates
+     *
+     * @param {Object} [facets] - the search facets object (defaults to `this.results.facets`)
+     */
+    getAggregates: function getAggregates(facets) {
+      var self = this;
+
+      return this.getStoredOptions(this.getQueryOptions()).then(function(data) {
+        var options = data.options.constraint;
+
+        facets = facets || self.results.facets;
+  
+        _.forIn( facets, function(facet, facetName) {
+          var facetType = facet.type;
+          
+          var myOption = options.filter(function (option) {
+            return option.name === facetName;
+          })[0];
+          if (!myOption) {
+            return $q.reject(new Error('No constraint exists matching ' + facetName));
+          }
+
+          var searchOptions = self.getQuery();
+          searchOptions.options = {};
+          searchOptions.options.constraint = [_.cloneDeep(myOption)];
+          if (myOption.range && myOption.range['facet-option']) {
+            myOption['values-option'] = myOption.range['facet-option'];
+            
+            // these work for all index types
+            myOption.aggregate = [
+              { apply: 'count' },
+              { apply: 'min' },
```

Comment:
Looks like you are only missing string in your list for min/max. And I'd expect anyUri to work as well..